src/js/utils.js

```diff
@@ -156,12 +156,24 @@ function estimateMaxEntropy(str) {
 
   // If all the characters come from one of these common character groups,
   // assume that the group is the domain of possible characters.
-  for (let chr_class in [BIN, DEC, HEX, ALPHA, ALPHANUM, B64, URL]) {
+  for (let chr_class of [BIN, DEC, HEX, ALPHA, ALPHANUM, B64, URL]) {
     let group = chr_class + SEPS;
     // Ignore separator characters when computing entropy. For example, Google
     // Analytics IDs look like "14103492.1964907".
-    if (str.split().every(val => group.includes(val))) {
+
+    // flag to check if each character of input string belongs to the group in question
+    let each_char_in_group = true;
+
+    [...str].forEach(char => {
```

Comment:
The `[...str]` converts the string to an array of chars, but we do this repeatedly, once per character class. Should save to a variable and reuse the variable instead.

Reply:
Ah, I see the reason for using `[...str]`: `.split('')` doesn't handle Unicode properly.

Reply:
Is unicode legal in URLs?

Reply:
I guess URLs are percent-encoded, so in this particular application `.split('')` should be OK.

Reply:
@bcyphers @ablanathtanalba Wait, `estimateMaxEntropy()` is also used with localStorage keys/values, so Unicode does matter. Plus, I just realized, there is no need to split at all, we can just iterate the string itself directly with `for...of`.

This should be fixed in 2213448632d7a6a44c5af0bb5d1fa2c96a91f908